### DISCUSSION TITLE
Error: MongoDB error, Kind: Server selection timeout: No available servers.

### DISCUSSION THREAD

**Discussion by feri-irawan** [prisma/prisma#21571] [Q&A] [answered]
### Question

I tried using prism with provider `mongodb` but when I tried `prisma.findMany()`, I got the following error:

```
Error: MongoDB error
Kind: Server selection timeout: No available servers. Topology: { Type: Single, Servers: [ { Address: cluster0.rqmvlnn.mongodb.net:27017, Type: Unknown, Error: Kind: I/O error: No such host is known. (os error 11001), labels: {} } ] }, labels: {}
   0: schema_core::commands::schema_push::Calculate `from`
             at schema-engine\core\src\commands\schema_push.rs:29
   1: schema_core::state::SchemaPush
             at schema-engine\core\src\state.rs:436
```

My .env:
```
DATABASE_URL=mongodb://myname:mypassword@host/mydb?retryWrites=true&w=majority&directConnection=true
```

My operating system is Window 11, i have also #11929 (directConnection=true) and `npx prisma db pull --force`

**Comment by ludralph** [comment, marked as answer]
Hi @feri-irawan 👋 

Thank you for raising this question.

Have you tried [whitelising](https://www.mongodb.com/docs/atlas/government/tutorial/allow-ip/) your IP address on MongoDB Atlas. Also are you able to connect directly using a GUI tool like MongoDB [Compass](https://www.mongodb.com/developer/products/compass/)?

**Reply by ChibuzoEkwue** [reply]
Having the same issue I have whitelisted my ip but still no result.

**Reply by ludralph** [reply]
Can you open a new[ discussion](https://github.com/prisma/prisma/discussions/new) @ChibuzoEkwue and document your issue using the template?

**Reply by ChibuzoEkwue** [reply]
Thank you. But I was able to finally solve it after going back and forth for days. Apparently if your computer time is wrong it can cause that error

**Comment by nicholas-masini** [comment]
For anyone else encountering the same issue, and have whitelisted your IP but still facing the same error, check your firewall and make sure you are allowing traffic on port 27017.
